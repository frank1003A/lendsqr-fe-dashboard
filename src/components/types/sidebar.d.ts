/**
 * This file allows me to reuse type declaration
 * that are needed in more than one component.
 */
export interface SideBarData {
    title: string;
    path: string;
    subNav: Array<subNav>;
}

export interface Image {
    src: string
}

export interface subNav extends Omit<SideBarData, 'subNav'> {
    icon?: Image["src"];
}