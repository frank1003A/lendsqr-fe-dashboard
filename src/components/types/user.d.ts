interface EDUCATION {
  level?: string;
  employmentStatus?: string;
  sector?: string;
  duration?: string;
  officeEmail?: string;
  monthlyIncome?: Array<string>;
  loanRepayment?: string;
}

interface SOCIALS {
  twitter?: string;
  facebook?: string;
  instagram?: string;
}

interface GUARANTOR {
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  emailaddress?: string;
  gender?: string;
  address?: string;
}

export interface PROFILE {
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  avatar?: string;
  gender?: string;
  bvn?: string;
  address?: string;
  currency?: string;
}

export interface USER {
  createdAt: string;
  orgName?: string;
  userName?: string;
  email?: string;
  lastActiveDate?: string;
  profile?: Profile;
  guarantor?: GUARANTOR;
  accountBalance?: string;
  accountNumber?: string;
  socials?: SOCIALS;
  education?: EDUCATION;
  id: string;
}

