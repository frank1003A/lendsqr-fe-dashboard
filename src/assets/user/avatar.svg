<svg width="100" height="100" viewBox="0 0 100 100" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle opacity="0.16" cx="50" cy="50" r="50" fill="#213F7D"/>
<path d="M36.0405 65.1796C36.4796 62.2202 37.7936 59.6264 39.9796 57.4C42.7405 54.6 46.0732 53.2 49.9796 53.2C53.886 53.2 57.2204 54.6 59.9796 57.4C62.1796 59.6266 63.5062 62.2204 63.9593 65.1796M58.1405 44.0204C58.1405 46.247 57.3468 48.1532 55.7593 49.7408C54.1734 51.3408 52.253 52.1408 50.0001 52.1408C47.7594 52.1408 45.8409 51.3408 44.2409 49.7408C42.6534 48.1533 41.8596 46.247 41.8596 44.0204C41.8596 41.7673 42.6534 39.8468 44.2409 38.2596C45.8409 36.6737 47.7596 35.8799 50.0001 35.8799C52.2532 35.8799 54.1737 36.6737 55.7593 38.2596C57.3468 39.8471 58.1405 41.7674 58.1405 44.0204Z" stroke="#213F7D" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
