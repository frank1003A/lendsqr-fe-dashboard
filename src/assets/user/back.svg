<svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M1.94997 15.3564C1.9945 15.4712 2.0613 15.5767 2.14684 15.6658L5.89684 19.4157C6.07263 19.5927 6.31285 19.6935 6.56248 19.6935C6.81211 19.6935 7.05232 19.5927 7.22812 19.4157C7.40508 19.24 7.50586 18.9997 7.50586 18.7501C7.50586 18.5005 7.40508 18.2603 7.22812 18.0845L5.07187 15.9376H27.6562C28.1742 15.9376 28.5937 15.5181 28.5937 15.0001C28.5937 14.4821 28.1742 14.0626 27.6562 14.0626H5.07187L7.22812 11.9158C7.5961 11.5478 7.5961 10.9525 7.22812 10.5845C6.86014 10.2165 6.26485 10.2165 5.89687 10.5845L2.14687 14.3345C2.06133 14.4236 1.99453 14.529 1.95 14.6439C1.90195 14.7564 1.87617 14.8771 1.875 15.0001C1.87617 15.1232 1.90195 15.2439 1.95 15.3564L1.94997 15.3564Z" fill="#545F7D"/>
</svg>
