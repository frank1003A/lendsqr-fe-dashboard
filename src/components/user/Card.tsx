import { formatNumberWithCommas } from "utils";

interface Props {
    /**image or icon to render on card: 40px */
    icon?: string;
    /**name or category of card */
    name: string;
    /**numbers */
    fig: number;
    /**id of component */
    id: string;
}

const Card = (props: Props) => {
  return (
    <div className="card" id={props.id}>
        <span className='icon'>
          <img src={props.icon} alt={`${props.icon}_icon`}/>
        </span>
        <span className='name'>{props.name}</span>
        <span className='fig'>{formatNumberWithCommas(props.fig)}</span>
    </div>
  )
}

export default Card