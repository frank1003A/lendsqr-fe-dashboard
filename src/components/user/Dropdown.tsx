interface Props {
  handleSelect: React.ChangeEventHandler<HTMLSelectElement>;
}

/**Component is only visible on mobile screens */

const Dropdown = ({ handleSelect }: Props) => {
  return (
    <select name="" id="" className="dropdown" onChange={handleSelect}>
      <option value={1}>General Details</option>
      <option value={2}>Documents</option>
      <option value={3}>Bank Details</option>
      <option value={4}>Loans</option>
      <option value={5}>Savings</option>
      <option value={6}>App and System</option>
    </select>
  );
};

export default Dropdown;
