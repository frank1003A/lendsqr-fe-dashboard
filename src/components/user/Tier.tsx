import { FaRegStar } from "react-icons/fa";

const Tier = () => {
  return (
    <div className="tier-star">
      <FaRegStar className="icon" />
      <FaRegStar className="icon" />
      <FaRegStar className="icon" />
    </div>
  );
};

export default Tier;
