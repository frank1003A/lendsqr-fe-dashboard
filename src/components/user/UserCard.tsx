import imgdefault from "assets/user/avatar.svg";
import { USER } from "components/types/user";
import Tier from "./Tier";

interface Props {
  user: USER;
  /**parent active button state */
  active: number;
  /**Callback Function */
  onButtonClick: (n: number) => void;
}

const UserCard = (props: Props) => {

  /**Handles CHanging between Default and User Avatar */
  const renderAvatar = (): string => {
    if (!props.user.profile.avatar) {
      return imgdefault
    }  else {
      return props.user.profile.avatar
    }
  }
  return (
    <div className="card1">
      <div className="general-info">
        <div className="username-img-id">
          <img src={renderAvatar()} alt="user_avatar" />
          <div className="name-id">
            <h1>{props.user.profile?.firstName} {props.user.profile?.lastName}</h1>
            <h2>{props.user.accountNumber}</h2>
          </div>
        </div>
        <hr />
        <div className="tier">
          <p>User's Tier</p>
          <Tier/>
        </div>
        <hr />
        <div className="current-trans">
          <h1 className="current-loan">â‚¦{props.user.accountBalance}</h1>
          <h2 className="bank-details">{props.user.profile?.bvn}/Providus Bank</h2>
        </div>
      </div>

      <div className="section-toggle">
        <button
          className={props.active === 1 ? "btn active" : "btn"}
          onClick={() => props.onButtonClick(1)}
        >
          General Details
        </button>
        <button
          className={props.active === 2 ? "btn active" : "btn"}
          onClick={() => props.onButtonClick(2)}
        >
          Documents
        </button>
        <button
          className={props.active === 3 ? "btn active" : "btn"}
          onClick={() => props.onButtonClick(3)}
        >
          Bank Details
        </button>
        <button
          className={props.active === 4 ? "btn active" : "btn"}
          onClick={() => props.onButtonClick(4)}
        >
          Loans
        </button>
        <button
          className={props.active === 5 ? "btn active" : "btn"}
          onClick={() => props.onButtonClick(5)}
        >
          Savings
        </button>
        <button
          className={props.active === 6 ? "btn active" : "btn"}
          onClick={() => props.onButtonClick(6)}
        >
          App and System
        </button>
      </div>
    </div>
  );
};

export default UserCard;
