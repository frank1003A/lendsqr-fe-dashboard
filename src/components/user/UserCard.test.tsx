/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UserCard from './UserCard';
import { USER } from 'components/types/user';

const user: USER = {
    id: "1",
  profile: {
    firstName: 'John',
    lastName: 'Doe',
    avatar: 'avatar.jpg',
    bvn: '12345678901'
  },
  accountNumber: '1234567890',
  accountBalance: '1000000',
  createdAt: new Date()
};

const onButtonClick = jest.fn();

describe('UserCard component', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    const { asFragment } = render(<UserCard user={user} active={1} onButtonClick={onButtonClick} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the user name and account number correctly', () => {
    const { getByText } = render(
      <UserCard
        active={1}
        user={user}
        onButtonClick={onButtonClick}
      />
    );
    const name = getByText(`${user.profile.firstName} ${user.profile.lastName}`);
    const accountNumber = getByText(`${user.accountNumber}`);
    expect(name).toBeInTheDocument();
    expect(accountNumber).toBeInTheDocument();
  });
});
