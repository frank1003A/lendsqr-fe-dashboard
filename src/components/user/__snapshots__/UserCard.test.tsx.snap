// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card1"
  >
    <div
      class="general-info"
    >
      <div
        class="username-img-id"
      >
        <img
          alt="user_avatar"
          src="avatar.jpg"
        />
        <div
          class="name-id"
        >
          <h1>
            John Doe
          </h1>
          <h2>
            1234567890
          </h2>
        </div>
      </div>
      <hr />
      <div
        class="tier"
      >
        <p>
          User's Tier
        </p>
        <div
          class="tier-star"
        >
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
        </div>
      </div>
      <hr />
      <div
        class="current-trans"
      >
        <h1
          class="current-loan"
        >
          â‚¦1000000
        </h1>
        <h2
          class="bank-details"
        >
          12345678901/Providus Bank
        </h2>
      </div>
      <select
        class="dropdown"
        id=""
        name=""
      >
        <option
          value="1"
        >
          General Details
        </option>
        <option
          value="2"
        >
          Documents
        </option>
        <option
          value="3"
        >
          Bank Details
        </option>
        <option
          value="4"
        >
          Loans
        </option>
        <option
          value="5"
        >
          Savings
        </option>
        <option
          value="6"
        >
          App and System
        </option>
      </select>
    </div>
    <div
      class="section-toggle"
    >
      <button
        class="btn active"
      >
        General Details
      </button>
      <button
        class="btn"
      >
        Documents
      </button>
      <button
        class="btn"
      >
        Bank Details
      </button>
      <button
        class="btn"
      >
        Loans
      </button>
      <button
        class="btn"
      >
        Savings
      </button>
      <button
        class="btn"
      >
        App and System
      </button>
    </div>
  </div>
</DocumentFragment>
`;
